bin/update_cucumber

```diff
@@ -77,13 +80,15 @@ class {
             && ($repository['package']['name'] ?? null) === 'cucumber/gherkin-monorepo'
             && isset($repository['package']['version'])
             && is_array($repository['package']['source'] ?? null)
-            && isset($repository['package']['source']['reference']),
+            && isset($repository['package']['source']['reference'])
+            && is_string($repository['package']['source']['reference']),
```

Comment:
Without the string check the returned array structure could theoretically contain `mixed` type, whereas we want specifically a string. A few other similar cases have been tackled below.

Reply:
I guess... I suppose it depends how much detail we want to go into for a CI script. Either it's going to fail on an assertion or it's going to blow up / produce an unexpected or invalid PR diff (which we'll see before merging) - not sure that it really matters which as ultimately the job will fail and need to be debugged?

Bearing in mind that we are parsing a file that is under our control and has a very defined schema...